modules/draco/src/workers/draco-writer-worker-node.ts

```diff
@@ -3,9 +3,9 @@ import '@loaders.gl/polyfills';
 import {WorkerBody, WorkerMessagePayload} from '@loaders.gl/worker-utils';
 import {DracoWriter} from '../draco-writer';
 
-(() => {
+(async () => {
   // Check that we are actually in a worker thread
-  if (!WorkerBody.inWorkerThread()) {
+  if (!(await WorkerBody.inWorkerThread())) {
```

Comment:
Why does this need to be async? Isn't it just about checking whether `self` is defined?

Reply:
I've added dynamic import instead of require. It is a promise

Reply:
Yes, but it seems to me that simply detecting if we are running in a worker shouldn't depend on that.

Reply:
It checks now `(await import('worker_threads')).parentPort`. Is there another way to check it for NodeJS?

Reply:
In the browser, the global variable `self` is defined in worker threads only. Not sure if that is also true in Node.js, but if it is, it should be a good non-async way.